### PR TITLE
[New Algorithm] Adding Myers Difference Algorithm

### PR DESCRIPTION
**Pull request by horita-yuya**
### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### Description
It seems that some algorithm finding shortest edit script are included in this. This Myers algorithm is same type algorithm. It improves calculation time and space, but simple.

Android's DiffUtil are implemented by Myers Algorithm. So it is practical.
To understand the mechanism is good and I hope this PR supports for that.

I also add the description about Edit Graph, it is useful for understand Myers Algorithm.

Thank you.

### CODE REVIEW

**Comment by h-yuya** [comment]
This is a reference. 
http://www.xmailserver.org/diff2.pdf

Thank you 🙏

File: Myers Difference Algorithm/README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+<img src='Images/EditGraph.png' height="400">
+
+The x-axis at the top of the graph represents one of the sequences, `X`. The y-axis at the left side of the graph represents the other sequence, `Y`. Hence, the two sequences in question is the following:
+
+```
+X = [A, B, C, A, B, B, A]
+Y = [C, B, A, B, A, C]
+```
+
+MDA generates the edit graph through the following steps:
```
Comment (kelvinlauKL) [inline Myers Difference Algorithm/README.md]: These steps are hard to understand. It seems worthwhile to explain what the blue line, green dots, dotted lines, and red number represents, just to give the reader more context before diving into the steps.

**Comment by kelvinlauKL** [comment]
It just occurred to me that we have an article on finding the longest common subsequence already: https://github.com/raywenderlich/swift-algorithm-club/tree/master/Longest%20Common%20Subsequence

Could you check it out to see if it makes sense to add your ideas to the current topic?

File: Myers Difference Algorithm/README.md (hunk 1 above)
Comment (h-yuya) [reply Myers Difference Algorithm/README.md]: Thank you for reviewing.

Yes, it is necessary. Understanding edit graph is important for understanding MDA.
I'll add more explanation 👍

**Comment by h-yuya** [comment]
I have known that there is an article for finding the LCS. (https://github.com/raywenderlich/swift-algorithm-club/tree/master/Longest%20Common%20Subsequence)

My topic, or MDA is faster than this algorithm. It runs for loop twice like below.
```swift
for i in self.characters {
 for j in other.characters {
 }
}
```
I takes O(N * M) times for the worst case 😱

**Comment by h-yuya** [comment]
One more, the LCS algorithm can get only LCS.
But, if someone understand my topic MDA and Edit Graph, it can be extended to getting Diff command, like insert or delete, not only LCS.

This may make it easy that , for example, difference refreshing for UITableView or UICollectionView.

**Comment by h-yuya** [comment]
@kelvinlauKL 
I mentioned about the difference between my article and LongestCommonSubsequence.
The major difference is calculation efficiency.

and I added some explanation about figure.
Thank you 🙏

**Comment by horita-yuya** [comment]
Please 🙏

**Comment by horita-yuya** [comment]
Anything else is required?

**Pull request merged**